packages/pluggableWidgets/tree-view-web/src/components/TreeView.tsx

```diff
@@ -0,0 +1,100 @@
+import { GUID } from "mendix";
+import { createElement, CSSProperties, HTMLAttributes, ReactElement, ReactNode, useState } from "react";
+import classNames from "classnames";
+
+import "../ui/TreeView.scss";
+
+interface TreeViewObject {
+    id: GUID;
+    value: string | undefined;
+    content: ReactNode;
+}
+
+export interface TreeViewProps {
+    class: string;
+    style?: CSSProperties;
+    items: TreeViewObject[];
+    isUserDefinedLeafNode: TreeViewBranchProps["isUserDefinedLeafNode"];
+    startExpanded: TreeViewBranchProps["startExpanded"];
+}
+
+export function TreeView({
+    class: className,
+    items,
+    style,
+    isUserDefinedLeafNode,
+    startExpanded
+}: TreeViewProps): ReactElement {
+    // TODO: for lazy loading/knowing whether there are children, it might be better to not render any DOM here if there are no items.
+    return (
+        <div className={classNames("mx-tree-view", className)} style={style}>
+            {items.map(({ id, value, content }) => (
+                <TreeViewBranch
+                    key={id}
+                    value={value}
+                    isUserDefinedLeafNode={isUserDefinedLeafNode}
+                    startExpanded={startExpanded}
+                >
+                    {content}
+                </TreeViewBranch>
```

Comment:
It would be nice to pass these as children to the TreeView component inside `packages/pluggableWidgets/tree-view-web/src/TreeView.tsx`. It prevents passing through an extra prop and looping through the items again.

Reply:
I thought we wanted to keep that file as a "connection between mendix and our react components"-component. This keeps it inline with that concept.

Reply:
We do not test entry points indeed, just the rendering component which is inside `components/`